fargate-task.tf

```diff
@@ -79,6 +80,10 @@ resource "aws_ecs_task_definition" "fargate" {
         file_system_id     = volume.value.file_system_id
         root_directory     = volume.value.root_directory
         transit_encryption = "ENABLED"
+        authorization_config {
+          access_point_id = volume.value.authorization_config.access_point_id
+          iam = volume.value.authorization_config.iam
+          }
```

Comment:
This block needs to by *dynamic* so that it is only applied when/if an *access_point_id* and *iam* field are present.

example would be something like:

dynamic "authorization_config" {
--for_each = try(volume.value.authorization_config.access_point_id, null) != null && try(volume.value.authorization_config.iam, null) != null) ? [1] : []
--content {
----access_point_id = volume.value.authorization_config.access_point_id
----iam = volume.value.authorization_config.iam
--}
}

In the end both conditions need to be met:
- deploy with *authorization_config*
- deploy without *authorization_config*